### PR TITLE
feat: support local preview, post split; add deploy preview

### PR DESCRIPTION
**Pull request by thiskevinwang**
# Description

This PR adds support for **local** & **deployed** previews, following the content split work from:
- https://github.com/hashicorp/terraform-website/pull/2228

## Local preview

This can be tested via...

```sh
make website/build-local && make website/local
```

- change some `mdx` file content, save and refresh the browser

## Deploy previews

- A new Vercel project (https://vercel.com/hashicorp/terraform) was created to create deploy previews for `terraform`
- Deploy preview here: https://terraform-git-kevin-local-preview-post-split-hashicorp.vercel.app/

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,23 +26,41 @@ staticcheck:
...
+	@docker run ${DOCKER_RUN_FLAGS} ${DOCKER_IMAGE} npm start
+
+website/local:
+	@echo "==> Starting website in Docker..."
+	@docker run ${DOCKER_RUN_FLAGS} ${DOCKER_IMAGE_LOCAL} npm start
+
+.PHONY: website/build-local
+website/build-local:
+	@echo "==> Building local Docker image"
+	@docker build https://github.com/hashicorp/terraform-website.git\#kevin/terraform-split \
```
Comment (thiskevinwang) [inline Makefile:58]: Apply this change when https://github.com/hashicorp/terraform-website/pull/2228 is merged
```suggestion
	@docker build https://github.com/hashicorp/terraform-website.git\#master \
```

File: website/scripts/website-build.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+if [ -d "$PREVIEW_DIR" ]; then
+    echo "$PREVIEW_DIR found"
+    CLONE_DIR="$PREVIEW_DIR-tmp"
+    from_cache=true
+fi
+
+# Clone the terraform-website project, if needed
+echo "⏳ Cloning the terraform-website repo, this might take a while..."
+git clone --depth=1 --branch=kevin/terraform-split https://github.com/hashicorp/$REPO_TO_CLONE.git "$CLONE_DIR"
```
Comment (thiskevinwang) [inline website/scripts/website-build.sh:17]: Apply this change when hashicorp/terraform-website#2228 is merged

```suggestion
git clone --depth=1 --branch=master https://github.com/hashicorp/$REPO_TO_CLONE.git "$CLONE_DIR"
```

**Comment by BRKalow** [comment]
@thiskevinwang what do you think about running the images through https://squoosh.app/ to optimize them? Some are quite large.

**Comment by thiskevinwang** [comment]
> @thiskevinwang what do you think about running the images through https://squoosh.app/ to optimize them? Some are quite large.

@BRKalow: Optimized them here: https://github.com/hashicorp/terraform/pull/30814/commits/aa2402055539307b12fe3b26a52a3fecd0c5e8ef

File: website/scripts/website-build.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+if [ -d "$PREVIEW_DIR" ]; then
+    echo "$PREVIEW_DIR found"
+    CLONE_DIR="$PREVIEW_DIR-tmp"
+    from_cache=true
+fi
+
+# Clone the terraform-website project, if needed
+echo "⏳ Cloning the terraform-website repo, this might take a while..."
+git clone --depth=1 --branch=master https://github.com/hashicorp/$REPO_TO_CLONE.git "$CLONE_DIR"
```
Comment (BRKalow) [inline website/scripts/website-build.sh:23]: We could remove the branch here, it'll clone the default branch by default.
```suggestion
git clone --depth=1 https://github.com/hashicorp/$REPO_TO_CLONE.git "$CLONE_DIR"
```

File: Makefile [hunk 4]
```diff
@@ -26,23 +26,42 @@ staticcheck:
...
 		--volume "$(shell pwd):/website/ext/terraform" \
+		--volume "$(shell pwd)/website:/website/preview" \
 		--publish "3000:3000" \
-		hashicorp/terraform-website:full \
-		npm start
+		-e "IS_CONTENT_PREVIEW=true" \
+		-e "PREVIEW_FROM_REPO=terraform" \
+		-e "NAV_DATA_DIRNAME=./preview/data" \
+		-e "CONTENT_DIRNAME=./preview/docs" \
+		-e "CURRENT_GIT_BRANCH=$$(git rev-parse --abbrev-ref HEAD)"
```
Comment (BRKalow) [inline Makefile:43]: style: It might make sense to put these at the top with the other variable declarations?

**Review by BRKalow — APPROVED** [review]
:shipit:

**Comment by thiskevinwang** [comment]
This should've been cherrypicked to `v1.1`

**Pull request merged**